controller/konnect/index_kongroute.go

```diff
@@ -31,3 +38,23 @@ func kongRouteUsesPlugins(object client.Object) []string {
 	}
 	return annotations.ExtractPluginsWithNamespaces(route)
 }
+
+func kongRouteRefersToKongService(object client.Object) []string {
+	route, ok := object.(*configurationv1alpha1.KongRoute)
+	if !ok {
+		return nil
+	}
+	svcRef := route.Spec.ServiceRef
+	if svcRef == nil ||
+		svcRef.Type != configurationv1alpha1.ServiceRefNamespacedRef ||
+		svcRef.NamespacedRef == nil {
+		return nil
+	}
+
+	namespace := route.Namespace
+	if svcRef.NamespacedRef.Namespace != "" {
+		namespace = svcRef.NamespacedRef.Namespace
+	}
```

Comment:
We might leave a similar note regarding the namespaced references support here.

Reply:
Good call. I was thinking about it but eventually I believe we should settle on removing the `Namespace` field from `ServiceRef` (which is what https://github.com/Kong/kubernetes-configuration/pull/107 does) and track the implementation of those in https://github.com/Kong/kubernetes-configuration/pull/106).

ac6dd7bc98403cfab39af774e1631b4ec9badd84 adds the comment about that.

WDYT?